### PR TITLE
feat(v2/design-system): add custom Tags theme and components

### PR DESCRIPTION
**Pull request by karrui**
## Problem

This PR adds the `Tag` primitive components. Will be mostly used in rendering selected options in the `multiselect` component.

## Solution

**Features**:

- add custom Tag components and theme

**Improvements**:

- update (badge theme to be better exportable (by not strict typing the exported theme)

## Before & After Screenshots
In storybook

### CODE REVIEW

File: frontend/src/theme/utils/constrast.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+// A bunch of utility functions to help with calculating the contrast ratio of two colours.
```
[the remark this answers is not in the document]
Comment (karrui) [reply frontend/src/theme/utils/constrast.ts:1]: Tbh, I really just used co-pilot to generate those (then verified that it's exactly what axe (the a11y checker) uses LOL)

https://github.com/dequelabs/axe-core/blob/develop/lib/checks/color/color-contrast-evaluate.js

[the remark this answers is not in the document]
Comment (mantariksh) [reply frontend/src/theme/utils/constrast.ts:1]: can link this in a comment then!

[the remark this answers is not in the document]
Comment (karrui) [reply frontend/src/theme/utils/constrast.ts:1]: I'll add more comments to the magic numbers too whilst I'm at it, thanks for raising this up!

[the remark this answers is not in the document]
Comment (karrui) [reply frontend/src/theme/utils/constrast.ts:1]: and fix the typo contrast -> constrast too much typing `const` that my fingers just muscle memory and butcher that filename lol

**Comment by karrui** [comment]
Will fix, my refactor failed the contrast

**Pull request merged**